openapi.yaml

```diff
@@ -965,6 +938,60 @@ paths:
                                 $ref: '#/components/schemas/Error'
                             example: 
                                 errors: ["An unexpected error occured"]    
+    /projects/{project_id}/notepad:         
+        put:
+            summary: 'Update the project notepad'
+            operationId: updateProjectNotepad
+            tags:
+                - Projects
+            parameters:
+              - name: project_id
+                in: path
+                description: ID of the Project
+                required: true
+                schema:
+                    type: string
+                    format: id
+            requestBody:
+                required: true
```

Comment:
I might consider making this `false`. If I `params.require(:notepad)` for this controller action but then send in a body of `{ "notepad": "" }`, I get a 400 bad request, which in our case, we hope to just "delete" the project note by passing an empty string

Reply:
Yup! So in order to make the notepad property itself required the schema would look like this:

```
            requestBody:
                required: true
                content:
                    'application/json':
                        schema:
                            required:
                              - notepad
                            properties:
                                notepad:
                                    type: string
                                    example: "An updated dexcription of the project"
```

That looks like this in the docs then

<img width="504" alt="Screen Shot 2022-09-30 at 1 36 20 PM" src="https://user-images.githubusercontent.com/854689/193335087-f2a8cae9-fbba-4f56-9cde-af5a037d4659.png">